model/version/db.go

```diff
@@ -127,6 +127,20 @@ func ByProjectId(projectId string) db.Q {
 		})
 }
 
+// ByProjectIdActivated finds all activated non-patch versions within a project.
+func ByProjectIdActivated(projectId string) db.Q {
+	return db.Query(
```

Comment:
Sorry to have you change this again, but I think there are inconsistencies with how builds are marked active where this doesn't work the same way as the waterfall. Specifically there are scenarios (not sure how) where builds that ran tasks aren't marked as active. The waterfall instead finds all versions and checks if there is at least 1 task that ran.

What I would do is move the fetchVersionsAndAssociatedBuilds and anyActiveTasks functions from waterfall.go to the model package and call them from both this route and the waterfall

Reply:
These function are not enough to get active versions, `getVersionsAndVariants` should also be moved, which couldn't be easily done, because it uses some waterfall specific types. Here what I think: we could query all versions, join builds and check if builds have active tasks. It's possible with newer mongo api (`$lookup`) without making sequnce of separate queries. Something like this:
```
db.versions.aggregate([
  {$lookup: {from: 'builds', as: 'mbuilds', localField: '_id', foreignField: 'version'}},
  {$project: {_id: 1, create_time: 1, mbuilds: 1}},
  {$unwind: '$mbuilds'},
  {$unwind: '$mbuilds.tasks'},
  {$match: {'mbuilds.tasks.a': true}}, {$project: {_id: 1, create_time: 1}}, {$sort: {create_time: 1}}, 
  {$group: {_id: '$_id'}}
]).pretty()
```